### PR TITLE
feat!: replace autoware_auto_msgs with autoware_msgs for planning modules

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Description

This is a subset of https://github.com/autowarefoundation/autoware.universe/pull/6893 to make it easier to review.
This includes all the modification for `planning` directory in the original PR. 

## Related links
https://github.com/autowarefoundation/autoware.universe/pull/6893
https://github.com/orgs/autowarefoundation/discussions/3862

## Interface changes
Message types are modified according to the table in [this comment](https://github.com/orgs/autowarefoundation/discussions/3862#discussioncomment-9638820).

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_intersection_module/src/util_type.hpp
Comment (mitsudome-r) [inline planning/behavior_velocity_intersection_module/src/util_type.hpp:1]: This file doesn't exist in main branch any more. This should be removed.

Comment (mitsudome-r) [reply planning/behavior_velocity_intersection_module/src/util_type.hpp:1]: I removed the file.

**Comment by mitsudome-r** [comment]
This PR is ready to merge now.

**Comment by yukkysaito** [comment]
no _auto_msgs by grep in vehicle component.

**Comment by xmfcx** [comment]
Depends on:
- https://github.com/autowarefoundation/autoware.universe/pull/7240

Checking that now.

**Review by xmfcx — APPROVED** [review]
`colcon build --symlink-install  --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=1  --packages-up-to   autoware_behavior_path_dynamic_obstacle_avoidance_module autoware_behavior_path_external_request_lane_change_module autoware_behavior_path_static_obstacle_avoidance_module autoware_behavior_velocity_planner autoware_behavior_velocity_template_module autoware_motion_velocity_out_of_lane_module autoware_motion_velocity_planner_common autoware_motion_velocity_planner_node autoware_path_optimizer autoware_planning_test_manager autoware_remaining_distance_time_calculator autoware_static_centerline_generator autoware_velocity_smoother behavior_path_avoidance_by_lane_change_module behavior_path_goal_planner_module behavior_path_lane_change_module behavior_path_planner behavior_path_planner_common behavior_path_sampling_planner_module behavior_path_side_shift_module behavior_path_start_planner_module behavior_velocity_blind_spot_module behavior_velocity_crosswalk_module behavior_velocity_detection_area_module behavior_velocity_dynamic_obstacle_stop_module behavior_velocity_intersection_module behavior_velocity_no_drivable_lane_module behavior_velocity_no_stopping_area_module behavior_velocity_occlusion_spot_module behavior_velocity_out_of_lane_module behavior_velocity_planner_common behavior_velocity_run_out_module behavior_velocity_speed_bump_module behavior_velocity_stop_line_module behavior_velocity_traffic_light_module behavior_velocity_virtual_traffic_light_module behavior_velocity_walkway_module bezier_sampler costmap_generator external_velocity_limit_selector freespace_planner freespace_planning_algorithms frenet_planner mission_planner objects_of_interest_marker_interface obstacle_cruise_planner obstacle_stop_planner obstacle_velocity_limiter path_sampler path_smoother planning_test_utils planning_topic_converter planning_validator route_handler rtc_interface rtc_replayer sampler_common scenario_selector surround_obstacle_checker`
was successful.

No `autoware_auto_x_msgs` references were left.

**Pull request merged**